syntax = "proto3";

service Cal{
    rpc add_matrix (request_msg) returns (response_msg) {};
    rpc mul_matrix (request_msg) returns (response_msg) {};
}

message row{
    repeated int32 value = 1;
}

message matrix{
    repeated row row = 1;
}

message request_msg{
    matrix matrix_1 = 1;
    matrix matrix_2 = 2;
}

message response_msg{
    matrix result = 1;
}